pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProviderToken.java

```diff
@@ -266,7 +301,7 @@ private Key getValidationKey(ServiceConfiguration conf) throws IOException {
             final byte[] validationKey = AuthTokenUtils.readKeyFromUrl(tokenPublicKey);
             return AuthTokenUtils.decodePublicKey(validationKey, publicKeyAlg);
         } else {
-            throw new IOException("No secret key was provided for token authentication");
+            return null;
```

Comment:
I don't think it's the third case, it seems like it should be considered adding it to the if branch of the publickey, which is essentially where the public key is read (file, string or an http host)
I think the `TokenSigningKeyResolver` should be called in the `readKeyFromUrl` function

Reply:
See lines 158-162.

This method just gets the public or secret key. When returns null, we read the JWKS data.